scripts/test/shared.py

```diff
@@ -281,6 +281,60 @@ def remove_readonly_and_try_again(func, path, exc_info):
     pass
 
 
+# This is a workaround for https://bugs.python.org/issue9400
+class Py2CalledProcessError(subprocess.CalledProcessError):
+  def __init__(self, returncode, cmd, output=None, stderr=None):
+    super(Exception, self).__init__(returncode, cmd, output, stderr)
+    self.returncode = returncode
+    self.cmd = cmd
+    self.output = output
+    self.stderr = stderr
+
+
+# https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess
+class Py2CompletedProcess:
+  def __init__(self, args, returncode, stdout, stderr):
+    self.args = args
+    self.returncode = returncode
+    self.stdout = stdout
+    self.stderr = stderr
+
+  def __repr__(self):
+    _repr = ['args=%s, returncode=%s' % (self.args, self.returncode)]
+    if self.stdout is not None:
+      _repr += 'stdout=' + repr(self.stdout)
+    if self.stderr is not None:
+      _repr += 'stderr=' + repr(self.stderr)
+    return 'CompletedProcess(%s)' % ', '.join(_repr)
+
+  def check_returncode(self):
+    if self.returncode != 0:
+      raise Py2CalledProcessError(returncode=self.returncode, cmd=self.args,
+                                  output=self.stdout, stderr=self.stderr)
+
+
+def run_process(cmd, check=True, input=None, universal_newlines=True,
+                capture_output=False, *args, **kw):
+  kw.setdefault('universal_newlines', True)
+
+  if hasattr(subprocess, "run"):
+    ret = subprocess.run(cmd, check=check, input=input, *args, **kw)
+    return ret
+
+  # Python 2 compatibility: Introduce Python 3 subprocess.run-like behavior
+  if input is not None:
+    kw['stdin'] = subprocess.PIPE
+  if capture_output:
+    kw['stdout'] = subprocess.PIPE
+    kw['stderr'] = subprocess.PIPE
+  proc = subprocess.Popen(cmd, *args, **kw)
+  stdout, stderr = proc.communicate(input)
+  result = Py2CompletedProcess(cmd, proc.returncode, stdout, stderr)
+  if check:
+    result.check_returncode()
+  return result
```

Comment:
This is all copied from emscripten's shared.py. The only difference is that I added support for the `capture_output` parameter in `run_process`.

Reply:
Should we add `capture_output` to emscripten's shared.py? (Not knowing what that enables in any detail)

Reply:
It makes me sad that we need all this boilerplate..

Reply:
It would be great to upgrade to python3 instead, but that should probably be a separate discussion.

Reply:
@jgravelle-google I think we can wait to add `capture_output` to emscripten's shared.py until someone wants to use it. All it does is store the process's stdout and stderr on the returned object so they can be programmatically inspected. Emscripten tends to store output in temporary files instead.